### PR TITLE
Make the RunnerManager and the SecretUpdater the same structure

### PR DESCRIPTION
**Pull request by masa213f**
The RunnerManager and the SecretUpdater are something similar.
They create one goroutine for each RunnerPool CR and do some functions.
So they can be made the same structure.

In this PR, I fixed the initialize method for them and some minor changes.
I did not change their behavior.

Signed-off-by: Masayuki Ishii <masa213f@gmail.com>

### CODE REVIEW

**Review by zoetrope — APPROVED** [review]

File: controllers/runner_manager_test.go [hunk 1]
```diff
@@ -335,13 +335,13 @@ var _ = Describe("RunnerManager", func() {
 
 		By("checking metrics are not exposed")
 		MetricsShouldNotExist(metricsURL, "meows_runnerpool_replicas")
-		MetricsShouldNotExist(metricsURL, "meows_runner_online")
+		// MetricsShouldNotExist(metricsURL, "meows_runner_online")
```
Comment (kmdkuk) [inline controllers/runner_manager_test.go:338]: Why is it commented out here?
I have confirmed that test panics if I don't comment out this section, but I couldn't figure out why.

Comment (masa213f) [reply controllers/runner_manager_test.go:338]: Thank you!
It was commented out for debugging. And I forgot to turn it off.

Comment (masa213f) [reply controllers/runner_manager_test.go]: @kmdkuk 
Could you try `git pull --force`?
Before I did force-push yesterday, I faced a panic too. Now the bug is fixed.

Comment (kmdkuk) [reply controllers/runner_manager_test.go]: Thank you.
I've confirmed that the test works in my local environment!

**Review by kmdkuk — APPROVED** [review]

**Pull request merged**